ci/fetch_artifacts.py

```diff
@@ -0,0 +1,145 @@
+import json
+import os
+import re
+import subprocess
+import requests
+import zipfile
+from datetime import datetime
+
+
+# This script is used to download the latest tagged build artifacts from the GitLab CI pipeline.
+class ArtifactsDownloader:
+    def __init__(
+        self, target_os, target_arch, token, download_dir="./", tag_prefix="nightly"
+    ):
+        self.target_os = target_os
+        self.target_arch = target_arch
+        self.token = token
+        self.download_dir = download_dir
+        self.tag_prefix = tag_prefix
+
+    def download(self) -> bool:
+        _, tag_msg = self._get_latest_tag(self.tag_prefix)
+
+        if tag_msg:
+            tag_json = json.loads(tag_msg)
+            return self._get_pipeline_build_artifacts(
+                tag_json["pipeline_id"],
+                self.download_dir,
+                self.target_arch,
+                self.target_os,
```

Comment:
There are redundant since the object already have these attributes.

```suggestion
                self.download_dir,
                self.target_arch,
                self.target_os,
```

Reply:
other than that +2